From 4112936f5a0c61147884c048aa9e8ec8a2c4968a Mon Sep 17 00:00:00 2001
From: Daniel Sangorrin <dsl@ertl.jp>
Date: Fri, 14 Dec 2012 14:38:27 +0900
Subject: [PATCH] QEMU fix arm gic targets

---
 hw/arm_gic.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/hw/arm_gic.c b/hw/arm_gic.c
index 2a19d3b..522193f 100644
--- a/hw/arm_gic.c
+++ b/hw/arm_gic.c
@@ -597,7 +597,7 @@ static void gic_dist_writeb(void *opaque, target_phys_addr_t offset,
           value = 0xff;
         for (i = 0; i < 8; i++) {
             if (value & (1 << i)) {
-                int mask = (irq < GIC_INTERNAL) ? (1 << cpu) : GIC_TARGET(irq);
+                int mask = (irq < GIC_INTERNAL) ? (1 << cpu) : GIC_TARGET(irq + i);
 
                 /* NOTE: TrustZone: Secure interrupt is RAZ/WI for
                  * normal world */
@@ -652,7 +652,7 @@ static void gic_dist_writeb(void *opaque, target_phys_addr_t offset,
                 /* NOTE: TrustZone: Secure interrupt is RAZ/WI for
                  * normal world */
                 if (!GIC_TEST_SECURE(irq + i, cm) || gic_is_secure_access(s)) {
-                    GIC_SET_PENDING(irq + i, GIC_TARGET(irq));
+                    GIC_SET_PENDING(irq + i, GIC_TARGET(irq + i));
                 } else {
                     DPRINTF("Reject non-secure set pending of IRQ %d\n",
                             irq + i);
-- 
1.7.9.5

